### PR TITLE
[#1755] Added remix button functionality

### PR DESCRIPTION
**Pull request by dseif**

### CODE REVIEW

File: src/ui/widget/controls.js [hunk 1]
```diff
@@ -78,6 +79,13 @@ define( [ "util/lang", "text!layouts/controls.html" ],
         }
       };
 
+      function remixInButter() {
+
+          var loc = window.location;
```
Comment (cadecairos) [inline src/ui/widget/controls.js]: indentation :D

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/c0b779eaf460ce2/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/c0b779eaf460ce2/output.txt

Total script time: 1.15 mins

Server is running on http://pr875.butyr.org/templates/basic/
Run the Test Suite: http://pr875.butyr.org/test/

File: src/ui/widget/controls.js [hunk 2]
```diff
@@ -78,6 +79,16 @@ define( [ "util/lang", "text!layouts/controls.html" ],
         }
       };
 
+      function remixInButter() {
+
+        var loc = window.location;
+        // webkit has location.origin but Firefox and Opera do not, make sure we handle them accordingly
+        loc.origin = loc.origin || ( loc.protocol + "//" + loc.host );
```
Comment (jbuck) [inline src/ui/widget/controls.js:118]: This feels too fragile; I think it'd make more sense for Cornfield to create this URL. Maybe you can just copy the Referrer header?

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brett-etherworks received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/60256b2a699f82d/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/60256b2a699f82d/output.txt

Total script time: 3.00 mins

File: src/embed.js [hunk 3]
```diff
@@ -54,6 +54,9 @@ function init( window, document ) {
       show( "#share" );
     }, false );
 
+    $( "#remix-post" ).addEventListener( "click", function() {
+      popcorn.emit( "remixInButter" );
+    });
```
Comment (k88hudson) [inline src/embed.js:59]: prevent bubbling?

File: src/ui/widget/controls.js [hunk 4]
```diff
@@ -308,6 +319,11 @@ define( [ "util/lang", "text!layouts/controls.html" ],
         volumechange();
       }
 
+      p.on( "remixInButter", remixInButter );
+      if ( remixButton ) {
+        remixButton.addEventListener( "click", remixInButter, false);
```
Comment (k88hudson) [inline src/ui/widget/controls.js:324]: spacing

File: cornfield/lib/user.js [hunk 5]
```diff
@@ -99,6 +99,24 @@ module.exports = {
       callback( err, doc.projects.id( pid ) );
     });
   },
+  findUser: function findUser( pid, callback ) {
+    if ( !pid ) {
+      callback( 'not enough parameters to search' );
```
Comment (k88hudson) [inline cornfield/lib/user.js:104]: double quotes

File: cornfield/lib/user.js [hunk 6]
```diff
@@ -99,6 +99,24 @@ module.exports = {
...
+      return;
+    }
+    UserModel.find({
+      "projects._id": pid
+    }).select( "projects.data" ).exec(function( err, doc ) {
```
Comment (k88hudson) [inline cornfield/lib/user.js:109]: spacing

**Pull request closed without merging**